Automattic/jetpack
# WP JS DataSync: Improve error resistance

## Proposed changes:
- Remove casting to specific types in favor of inferred types from the useDataSync hook.
- Protect against missing window values by adding try-catch around the initialData callback.
- Include `ds-debug-disable` parameter in URLs for debugging purposes.
- Use `ds-debug-disable` to disable data synchronization or action handling based on URL hash or GET parameter.

### Other information:



## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Navigate to the Boost Dashboard and append the `#ds-debug-disable=<key>` or `?ds-debug-disable=<key>` to see if the corresponding DataSync values or action routes are disabled as expected.
- Confirm that updating the hash in the browser's address bar affects the subsequent DataSync requests appropriately.
- Ensure that no TypeScript type errors occur due to the removal of casting. All data should be correctly inferred from the useDataSync hook.
- Check that the application recovers gracefully when values are missing.
